### PR TITLE
Fixes 500 error on stock transfer with no variant

### PR DESCRIPTION
**Pull request by ted**
Fixes issue stated in #8253

### CODE REVIEW

File: backend/app/controllers/spree/admin/stock_transfers_controller.rb [hunk 1]
```diff
@@ -19,18 +19,22 @@ def show
...
-        variants = Hash.new(0)
-        params[:variant].each_with_index do |variant_id, i|
-          variants[variant_id] += params[:quantity][i].to_i
+        if params[:variant].nil?
+          flash[:error] = Spree.t('stock_transfer.errors.must_have_variant')
+          render :new
+        else
+          variants = Hash.new(0)
+          params[:variant].each_with_index do |variant_id, i|
+            variants[variant_id] += params[:quantity][i].to_i
```
Comment (tanmay3011) [inline backend/app/controllers/spree/admin/stock_transfers_controller.rb]: Shouldn't we also consider rejecting the transfers which have 0 or nil as quantity as I don't find any reason to create a stock transfer with a zero quantity?
@damianlegawiec

**Comment by tanmay3011** [comment]
@tedmaloney Please squash commits

**Comment by tedmaloney** [comment]
I'm not sure how squashing my commits made the tests not pass when the passed before, could you rebuild it to double check?

**Comment by damianlegawiec** [comment]
Thanks @tedmaloney 👍 flaky specs sometimes cause problems on CI - we're working on it

**Pull request merged**